### ISSUE TITLE
Repeated augmentation layer

### ISSUE THREAD

**Issue by sayakpaul** [keras-team/keras-cv#372]
Repeated augmentation layer [1] has also become an important recipe to train SoTA image classification models. The abstract of [1] pretty much sums up what it is:

> Large-batch SGD is important for scaling training of deep neural networks. However, without fine-tuning hyperparameter schedules, the generalization of the model may be hampered. We propose to use batch augmentation: replicating instances of samples within the same batch with different data augmentations. Batch augmentation acts as a regularizer and an accelerator, increasing both generalization and performance scaling for a fixed budget of optimization steps. We analyze the effect of batch augmentation on gradient variance and show that it empirically improves convergence for a wide variety of networks and datasets. Our results show that batch augmentation reduces the number of necessary SGD updates to achieve the same accuracy as the state-of-the-art. Overall, this simple yet effective method enables faster training and better generalization by allowing more computational resources to be used concurrently.

**References**

[1] [Augment Your Batch: Improving Generalization Through Instance Repetition](https://openaccess.thecvf.com/content_CVPR_2020/papers/Hoffer_Augment_Your_Batch_Improving_Generalization_Through_Instance_Repetition_CVPR_2020_paper.pdf)

**Comment by bhack** [comment]
We have talked about this some month ago at  https://github.com/keras-team/keras-cv/pull/146#issuecomment-1048315498.

At that time we have not received any feedback.

**Comment by bhack** [comment]
P.s. We have still many performance  fallback related to the batch randomizzation strategy:
 https://github.com/keras-team/keras-cv/issues/291

**Comment by sayakpaul** [comment]
Makes sense to have a dedicated issue for the layer rather than keeping it within comments. That way it becomes more visible and all the related concerns (like the ones you mentioned) can be discussed specifically.

**Comment by bhack** [comment]
Probably, but as you can see we was in a design phase on Feb and comments emergerged in that context.
Now, at least, the history is complete also here for who is landing on this ticket directly.
